### ISSUE TITLE
bit_string_literal_500 & 501 not detected in some places

### ISSUE THREAD

**Issue by maltaisn** [jeremiah-c-leary/vhdl-style-guide#1449]
After applying the two rules on our code base, these three cases were not fixed:

1. When the literal is an argument to a procedure call inside a procedure.
2. When the literal is a port map assignment
3. When the literal is a case/when value

```vhdl
library ieee;
  use ieee.std_logic_1164.all;
  use ieee.numeric_std.all;

entity b is
  port (
    a : in    std_logic_vector(7 downto 0);
    b : out   std_logic_vector(7 downto 0)
  );
end entity b;

architecture a of b is

  procedure foo (
    a : std_logic_vector;
    b : boolean
  ) is
  begin

  end procedure foo;

  procedure bar (
    signal a : std_logic_vector
  ) is
  begin

    foo(X"5a", a = X"5a");  -- 1

  end procedure bar;

  component my_component is
    port (
      c : in    std_logic_vector(7 downto 0)
    );
  end component my_component;

begin

  inst : component my_component
    port map (
      c => X"5a"  -- 2
    );

  proc : process is
  begin

    case a is

      when X"5a" =>  -- 3

        null;

      when others =>

        null;

    end case;

  end process proc;

end architecture a;
```
This file is the result after applying the fixes with the default configuration, the expected literal being `x"5A"` and not `X"5a"`.

**Comment by JHertz5** [comment]
Hi @maltaisn.

Sorry for the delay! I've create a branch, `issue-1449`, that I believe resolves this issue. I've made changes to the parser to improve it's classification of bit string literals in the contexts that you described above. Please give the branch a try and let me know whether you're satisfied that the changes are sufficient.

**Comment by maltaisn** [comment]
Hi. I tested it on our code base and it seems to cover all cases.

However, I found this introduces a regression:
```vhdl
library ieee;
  use ieee.std_logic_1164.all;

entity a is
end entity a;

architecture behavioral of a is

  constant baz : baz_type :=
  (
    a => std_logic_vector(to_unsigned(1, BAZ_TYPE.a'length)),
    b => std_logic_vector(to_unsigned(2, BAZ_TYPE.b'length)),
    c => std_logic_vector(to_unsigned(3, BAZ_TYPE.c'length)),
    d => std_logic_vector(to_unsigned(4, BAZ_TYPE.d'length)),
    e => std_logic_vector(to_unsigned(5, BAZ_TYPE.e'length)),
    f => std_logic_vector(to_unsigned(6, BAZ_TYPE.f'length))
  );

  signal aBCD : std_logic_vector(0 downto 0);

begin

  b : process is
  begin

    foo(c(bar(0)) & c(aBCD(0)));    -- attribute_500 error here, "aBCD: gets fixed to "abcd"

  end process b;

end architecture behavioral;
```
With the following rules:
```yaml
rule:
  signal_004:
    disable: yes
```
There's an `attribute_500` error. This doesn't happen when I check out the base of the issue-1449 branch.

**Comment by JHertz5** [comment]
@maltaisn Thanks very much for this feedback! That's unexpected, but I can recreate your error. It's interesting that the VSG unit tests didn't catch this, I'll update them to include your code example.

It looks like the parser changes that I made have inadvertently broken the parsing of attributes. I'll have to give it another try.
